### ISSUE TITLE
Weighted NER Loss

### ISSUE THREAD

**Issue by atakanokan** [explosion/spaCy#4629]
## Feature description
Using different weights for NER classes to enable learning towards more important classes. For example if detecting PERSONs is much more important to my task than detecting LOCATIONs, I would love to configure the loss function so that it PERSON's contribution is much higher.

It can be a custom component that users can pass a dictionary or two lists that reweighs the loss function based on the users' needs. [custom component](https://spacy.io/usage/processing-pipelines#custom-components)

**Comment by svlandeg** [comment]
Hi @atakanokan, sorry for the very late reply. I understand your use-case, but we're not currently considering this feature for inclusion in the core library.

I think you could easily simulate this behaviour by just upsampling the "important" classes in your training dataset, which will automatically bias the ML model towards them.

If you really do want to adjust the loss function itself, you'll probably need to create a new subclass of `EntityRecognizer`, and adjust the loss function. The new [spaCy v3](https://github.com/explosion/spaCy/releases/tag/v3.0.0rc1) brings a lot of modularity, but we don't yet have a system to easily replace these loss functions, and they are often quite tightly coupled with the rest of the component's implementation.

More docs on how trainable components look like in v3, can be found here: https://nightly.spacy.io/usage/processing-pipelines#trainable-components

Good luck to you if you do wish to implement this yourself, and feel free to open a new issue if you run into problems!
